### PR TITLE
Add utility for toasting errors with fallback title [UI-149]

### PR DESCRIPTION
**Pull request by teamchong**
This PR introduces a toastError utility function to streamline error handling in toast notifications. The function simplifies error reporting by:

- Checking if the error is an HTTP error, extracting the title and message.
- Using a fallback title if details are unavailable.

Closes [UI-149](https://eventstore.aha.io/develop/features/UI-149)

### CODE REVIEW

File: packages/components/src/components/actions/ActionDelete.tsx [hunk 1]
```diff
@@ -67,18 +67,6 @@ const handleDeletion =
...
-                    message: details.detail,
-                });
-            } else {
-                toast.error({
-                    title: `Failed to delete ${description}`,
-                    message:
-                        error instanceof Error ? error.message : String(error),
-                });
-            }
+            toastError(error, `Failed to delete ${description}`);
```
Comment (George-Payne) [inline packages/components/src/components/actions/ActionDelete.tsx:70]: Looks like this is using a previous version?

File: packages/components/package.json [hunk 2]
```diff
@@ -58,6 +58,7 @@
         "@stencil-community/postcss": "^2.2.0",
         "@stencil/core": "4.22.0",
         "jest": "^27.4.5",
+        "jsdom": "^25.0.1",
```
Comment (George-Payne) [inline packages/components/package.json:61]: Do we need this?

File: packages/components/src/utils/toast.ts [hunk 3]
```diff
@@ -34,4 +35,21 @@ export const toast = {
      * Called when an error occured, to inform the user.
      */
     error: popToast('error'),
+    /**
+     * Handles HTTPError instances specifically and displays them as error toasts.
+     */
+    async errorWithDetails(err: unknown, fallbackTitle: string) {
```
Comment (George-Payne) [inline packages/components/src/utils/toast.ts:41]: Maybe `toast.httpError` is a better name. or `toast.fromError`.

Think it's mostly bad initial design, should probably have been:

```
toast({ level: "error", ...options });
```

then we could have used `toast.error` for this, but too late now without a bunch of breaking changes.

**Review by George-Payne — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 4]
```diff
@@ -4871,9 +4871,9 @@ __metadata:
   linkType: hard
 
 "decimal.js@npm:^10.2.1":
-  version: 10.2.1
-  resolution: "decimal.js@npm:10.2.1"
-  checksum: 2931d044a01b423f6f449c2cadbf6ee0e738096cde3f2d9b4c392b7c0cbdc5a41961855f67c1ba7dbabf439ecd76a923b495d100866984a7bcaac4e225f12fe9
+  version: 10.4.3
```
Comment (George-Payne) [inline yarn.lock:4874]: Lets revert the `yarn.lock` also, looks like these changes came from adding jsdom, but are now left in

**Review by George-Payne — CHANGES_REQUESTED** [review]

File: packages/components/src/utils/toast.ts (hunk 3 above)
Comment (teamchong) [reply packages/components/src/utils/toast.ts]: how about ``toast.error({ checkHttpError: true, fallbackTitle: `Failed to delete ${description}` })``

**Review by George-Payne — APPROVED** [review]

**Pull request merged**